core/ecschema-editing/src/Validation/ECRules.ts

```diff
@@ -453,15 +453,20 @@ export async function* validateNavigationProperty(property: AnyProperty): AsyncI
     yield new Diagnostics.NavigationRelationshipAbstractConstraintEntityOrMixin(property, [property.fullName, relationship.fullName]);
   }
 
-  let concreteClass = false;
+  let classSupported = false;
   if (thisConstraint.constraintClasses) {
     for (const constraintClass of thisConstraint.constraintClasses) {
       if (constraintClass.fullName === property.class.fullName)
-        concreteClass = true;
+        classSupported = true;
+      else {
+        const inheritedProp = await navProp.class.getInheritedProperty(navProp.name);
+        if (inheritedProp && constraintClass.fullName === inheritedProp.class.fullName)
```

Comment:
There could be N levels of property overrides.  This check will only work for one level of property override.